### ISSUE TITLE
[Performance] Convolution layer issue profiling

### ISSUE THREAD

**Issue by Apisteftos** [microsoft/onnxruntime#16926]
### Describe the issue

I am doing profiling and I am using the pre-trained Faster-RCNN model from the Model Zoo. 

### To reproduce

According to [NVIDIA](https://docs.nvidia.com/deeplearning/performance/dl-performance-matrix-multiplication/index.html#dim-quantization) we are measuring the FLOPs: 2* M * N * K where in the forward propagation:

M = N * P * Q (batch size * out height * out width)
N = K (number of the out Channels)
K = C * R * S   (input channel * kernel height * kernel width) 

Some convolutional layers i.e the 486 behaves strangely with big input and output data total different and gives me 788 TFLOPS, while my GPU supports theoretical peak 7 TFLOPs. Are these theoretical peaks not divided by the latency?

M = 1x200x304
N = 256
K = 256x3x3
Latency = 91 microseconds

But I didn't estimated according to NVIDIA, firstly I estimated the macs which is the output feature map 1x256x200x304 multiplied by M and then accumulated with the output feature map. The results are the same with both ways.

The latency of this layer during the inference is 91 microseconds. How is that possible to have more TFLOPs than the theoretical peak? After searching in the [onnxruntime](https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/providers/cuda/nn/conv.cc) I found that 8 possible optimization algorithms are used. Implicit Gemm, FFT, Winograd etc. I attached two profile nodes to understand this strange behavior.  

What is happening exactly is someone of these algorithms ? Are divide the output filters and then run the tiles parallel in the GPU? Then if that happens should I divide the output filter by a factor to get the real performance? 

### Urgency

I am having a strict project and I must understand this behavior. 

### Platform

Linux

### OS Version

Ubuntu 22.04

### ONNX Runtime Installation

Released Package

### ONNX Runtime Version or Commit ID

1.15.0

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

CUDA

### Execution Provider Library Version

CUDA 11.6

### Model File

[profiling_data.json.zip](https://github.com/microsoft/onnxruntime/files/12209623/profiling_data.json.zip)

### Is this a quantized model?

No
